lib/index.js

```diff
@@ -114,18 +101,52 @@ function setupRateLimitEventExt(server, options) {
   };
 
   server.ext(event, (request, reply) => {
+    const options = getRouteOptions(request);
+    if (!options.enabled) {
+      return reply.continue();
+    }
     // This handler is going to be called one time per registration of patova
-    getType(request, reply, (err, type) => {
-      extractKeyAndTakeToken(options.limitd, request, reply, type);
+    getType(options, request, reply, (err, type) => {
+      extractKeyAndTakeToken(options, request, reply, type);
     });
   });
+
+  server.ext('onPreResponse', (request, reply) => {
+    const requestLimit = request.plugins.patova && request.plugins.patova.limit;
+
+    if (requestLimit && requestLimit.conformant) {
+      const options = getRouteOptions(request);
+      if (options.sendResponseHeaders) {
+        const headers = new RateLimitHeaders(
+            requestLimit.limit,
+            requestLimit.remaining,
+            requestLimit.reset);
+
+        Object.keys(headers).forEach(
+          key => setResponseHeader(request, key, headers[key]));
+      }
+    }
+    reply.continue();
+  });
+
+  function getRouteOptions(request) {
+    const routeOptions = request.route.settings.plugins.patova;
+    if (routeOptions === undefined) {
+      return pluginOptions;
+    }
+    if (routeOptions._merged) { //already merged with previous pluginOptions;
```

Comment:
maybe this is a premature optimization? I think hapi [merges per request](https://github.com/hapijs/hapi/blob/832ed3d5d2c18c4f7f1cc8bfeca8cd1595ac09ed/lib/validation.js). Storing `_merged` on the `plugins.patova` object which we don't own feels odd.